### PR TITLE
adding infrastructure golden topology provisioning logic

### PR DESCRIPTION
**Pull request by vbudi000**
This pull request has pre-requisites of other pull requests:
- https://github.com/cloud-native-toolkit/multi-tenancy-gitops-infra/pull/11
- https://github.com/cloud-native-toolkit/toolkit-charts/pull/237

The modification will allow easy modification for adding Infrastructure and storage nodes, reallocate the pods and install OCS.... The pre-requisite for this is to enable in 0-bootstrap/single-cluster/1-infra/kustomization.yaml to enable:
```
- argocd/namespace-openshift-storage.yaml
- argocd/storage.yaml
- argocd/infraconfig.yaml
- argocd/machinesets.yaml
```

### CODE REVIEW

**Comment by vbudi000** [comment]
solved #26

**Review by hollisc — APPROVED** [review]
Looks good

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -150,6 +150,14 @@ fork_repos () {
 
 }
 
+check_infra () {
+   if [[ "${ADD_INFRA}" == "yes" ]]; then 
+     pushd ${OUTPUT_DIR}/gitops-0-bootstrap
+       bash ./scripts/infra-mod.sh
```
Comment (csantanapr) [inline scripts/bootstrap.sh:156]: Using `bash` does the environment variables get propagated to the infra-mod.sh script?

Comment (csantanapr) [reply scripts/bootstrap.sh:156]: I think we should consistency and use `source`
we already do in other places in the same script
https://github.com/cloud-native-toolkit/multi-tenancy-gitops/blob/master/scripts/bootstrap.sh#L352

**Review by csantanapr — CHANGES_REQUESTED** [review]
please sign-off DCO and address the execute to `bash` for `source`

File: scripts/bootstrap.sh (hunk 1 above)
Comment (vbudi000) [reply scripts/bootstrap.sh]: Updated to use `source`

**Review by csantanapr — APPROVED** [review]

**Pull request merged**